lib/spack/docs/packages_yaml.rst

```diff
@@ -407,9 +483,18 @@ like this:
 
    packages:
      all:
-       require: '%clang'
+       require: '%[when=%c]c=clang %[when=%cxx]cxx=clang'
+
+which means every spec will be required to use ``clang`` as the compiler for C and C++ code.
 
-which means every spec will be required to use ``clang`` as a compiler.
+.. warning::
+
+   The simpler config ``require: %clang`` will fail to build any
+   package that does not include compiled code, because those packages
+   cannot depend on ``clang`` (alias for ``llvm+clang``). In most
+   contexts, default requirements must use either conditional
+   dependencies or a :ref:`toolchain <toolchains>` that combines conditional
+   dependencies.
```

Comment:
I think we should revert this specific change (will submit a PR to do so). The fact is that we introduce how default requirements work by using the most complex syntax we have +  a big warning on not using `%`. It would probably be more sensible to either skip these few lines, or use a simple variant as an introductory example.